concourse-chrome-driver/terraform/deployments/670214072732/codepipeline.tf

```diff
@@ -0,0 +1,102 @@
+    location = data.aws_s3_bucket.artifact_store.bucket
+  }
+
+  stage {
+    name = "Source"
+
+    action {
+      name             = "Source"
+      category         = "Source"
+      owner            = "AWS"
+      provider         = "CodeStarSourceConnection"
+      version          = "1"
+      output_artifacts = ["git_base_image"]
+      configuration = {
+        ConnectionArn    = "arn:aws:codestar-connections:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:connection/${var.codestar_connection_id}"
+        FullRepositoryId = "alphagov/cyber-security-concourse-base-image"
+        BranchName       = "master"
+      }
+    }
+  }
+
+
+  stage {
+    name = "Changes"
+
+    action {
+      name             = "GetChangedFiles"
+      category         = "Build"
+      owner            = "AWS"
+      provider         = "CodeBuild"
+      version          = "1"
+      input_artifacts  = ["git_base_image"]
+      output_artifacts = ["changed_files"]
+      configuration = {
+        ProjectName = module.codebuild-get-changed-file-list.project_name
+      }
+    }
+  }
+
+  stage {
+    name = "Actions"
+
+    action {
+      name             = "GetActionsRequired"
+      category         = "Build"
+      owner            = "AWS"
+      provider         = "CodeBuild"
+      version          = "1"
+      input_artifacts  = ["git_base_image", "changed_files"]
+      output_artifacts = ["actions_required"]
+      configuration = {
+        PrimarySource = "git_base_image"
+        ProjectName   = module.codebuild-get-actions-required.project_name
+      }
+    }
+  }
+
+  stage {
+    name = "BuildDockerImages"
+
+    action {
+      name             = "BuildAndPushCdImage"
+      category         = "Build"
+      owner            = "AWS"
+      provider         = "CodeBuild"
+      version          = "1"
+      run_order        = 1
+      input_artifacts  = ["git_base_image", "actions_required"]
+      output_artifacts = []
+
+      configuration = {
+        PrimarySource        = "git_base_image"
+        ProjectName          = module.codebuild-build-container-docker-hub-cd.project_name
+        EnvironmentVariables = jsonencode([{ "name" : "CHECK_TRIGGER", "value" : 1 }, { "name" : "ACTION_NAME", "value" : "BuildAndPushCdImage" }])
+      }
+    }
+
+    action {
+      name             = "BuildAndPushConcourseImage"
+      category         = "Build"
+      owner            = "AWS"
+      provider         = "CodeBuild"
+      version          = "1"
+      run_order        = 1
+      input_artifacts  = ["git_base_image", "actions_required"]
+      output_artifacts = []
+
+      configuration = {
+        PrimarySource        = "git_base_image"
+        ProjectName          = module.codebuild-build-container-docker-hub-concourse.project_name
+        EnvironmentVariables = jsonencode([{ "name" : "CHECK_TRIGGER", "value" : 1 }, { "name" : "ACTION_NAME", "value" : "BuildAndPushCdImage" }])
+      }
+    }
+  }
```

Comment:
Need to add a stage to apply the pipeline terraform

Reply:
do we have an example of where this is used ? Thanks.